### ISSUE TITLE
HTTP transport fails to decode messages

### ISSUE THREAD

**Issue by rkervella** [BishopFox/sliver#540]
The implant doesn't properly decodes the `execute-assembly` task in beacon mode over HTTP. The following commands results in an error on the implant side in the HTTP client:

```shell
[server] sliver (AFRICAN_KITTY) > execute-assembly /tmp//Rubeus.exe /help
```

On the implant side, here are the relevant debug lines:

```
...
2021/11/09 02:34:37 sliver.go:208: [beacon] sending check in ...
2021/11/09 02:34:37 httpclient.go:521: [http] segments = [db oauth2 namespaces php oauth api api], filename = index, ext = php
2021/11/09 02:34:37 httpclient.go:329: [http] POST -> http://172.16.10.218:8000/db/oauth2/namespaces/php/oauth/api/api/index.php?r=8a0228086 (76 bytes)
2021/11/09 02:34:37 httpclient.go:335: [http] POST request completed
2021/11/09 02:34:37 sliver.go:221: [beacon] recv task(s) ...
2021/11/09 02:34:37 httpclient.go:521: [http] segments = [script], filename = route, ext = js
2021/11/09 02:34:37 httpclient.go:251: [http] GET -> http://172.16.10.218:8000/script/route.js?c=629123b43
2021/11/09 02:34:37 httpclient.go:277: Decoding failed unexpected EOF
2021/11/09 02:34:37 sliver.go:226: [beacon] recv failure unexpected EOF
2021/11/09 02:34:37 sliver.go:203: [beacon] closing ...
2021/11/09 02:34:37 httpclient.go:521: [http] segments = [], filename = favicon, ext = png
2021/11/09 02:34:37 httpclient.go:375: [http] GET -> http://172.16.10.218:8000/favicon.png?s=5549h69o6
2021/11/09 02:34:37 httpclient.go:391: [http] session closed
...
```
The decoding error seems to happen in `SliverHTTPClient.ReadEnvelope()` here:

https://github.com/BishopFox/sliver/blob/03d1f92441f3a0105abdeccec1596bda56fcc55c/implant/sliver/transports/httpclient/httpclient.go#L274-L281

I'm not sure if it's related to the size of the message (since we're sending an assembly file of multiple KB, even MB) or not.

This was tested on both the `v1.5.0_beacon` and `stage` branches, with the same behavior.

**Comment by rkervella** [comment]
Same thing seems to happen with `execute-shellcode` when I send a sliver shellcode, so I think it's related to the size of the message.

**Comment by rkervella** [comment]
`upload` also seem to break with:

```
2021/11/09 03:51:49 session.go:288: [http] error: &errors.errorString{s:"chacha20poly1305: message authentication failed"}
2021/11/09 03:51:49 session.go:230: [http] lost connection, cleanup...
```

**Comment by rkervella** [comment]
Seems like the client doesn't read/receive all the data:

```
2021/11/09 04:43:06 httpclient.go:275: envelope size: 8008
```

Here's the server log:

```
DEBU[2021-11-09T13:43:06+01:00] [sliver/server/c2/http.go:576] envelope size: 766218 
DEBU[2021-11-09T13:43:06+01:00] [sliver/server/c2/http.go:586] encoded data size: 1021772 
```

**Comment by moloch--** [comment]
This may be related to the introduction of limited i/o readers to prevent DOS in v1.5

**Comment by rkervella** [comment]
@cha0tic87 this has been fixed in 1.5
